ledger-core/virtual/execute/execute_test.go

```diff
@@ -488,3 +493,69 @@ func TestSMExecute_VCallResultPassedToSMObject(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestSendVStateReportWithMissingState_IfConstructorWasInterruptedBeforeRunnerCall(t *testing.T) {
+	t.Log("C5084")
+	var (
+		mc  = minimock.NewController(t)
+		ctx = context.Background()
+
+		class                              = gen.UniqueGlobalRef()
+		caller                             = gen.UniqueGlobalRef()
+		catalog     object.Catalog         = object.NewLocalCatalog()
+		authService authentication.Service = authentication.NewServiceMock(t)
+	)
+
+	slotMachine := slotdebugger.New(ctx, t)
+	slotMachine.PrepareRunner(ctx, mc)
+
+	slotMachine.AddInterfaceDependency(&catalog)
+	slotMachine.AddInterfaceDependency(&authService)
+
+	var vStateReportRecv = make(chan struct{})
+	slotMachine.PrepareMockedMessageSender(mc)
+	slotMachine.MessageSender.SendRole.Set(func(_ context.Context, msg payload.Marshaler, role node.DynamicRole, object reference.Global, pn pulse.Number, _ ...messageSender.SendOption) error {
+		res, ok := msg.(*payload.VStateReport)
+		require.True(t, ok)
+		require.Equal(t, payload.Missing, res.Status)
+		close(vStateReportRecv)
+		return nil
+	})
+
+	outgoing := gen.UniqueLocalRefWithPulse(slotMachine.PulseSlot.CurrentPulseNumber())
+
+	smExecute := SMExecute{
+		Payload: &payload.VCallRequest{
+			CallType:     payload.CTConstructor,
+			CallFlags:    payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+			CallOutgoing: outgoing,
+
+			Caller:         caller,
+			Callee:         class,
+			CallSiteMethod: "New",
+		},
+		Meta: &payload.Meta{
+			Sender: caller,
+		},
+	}
+	slotMachine.Start()
+	defer slotMachine.Stop()
+
+	smWrapper := slotMachine.AddStateMachine(ctx, &smExecute)
+	slotMachine.RunTil(smWrapper.BeforeStep(smExecute.stepStartRequestProcessing))
+	slotMachine.Migrate()
```

Comment:
It's a pulse increment, isn't it?

Reply:
Yes, but this is a semi-integration test without server. Pulse number does not matter in this case. We test that migration send message